ansible/modules/hashivault/hashivault_secret_engine.py

```diff
@@ -73,6 +112,24 @@ def main():
         module.exit_json(**result)
 
 
+def parse_duration(duration):
+    regex = re.compile(r'((?P<days>\d+)d)|((?P<hours>\d+)h)|((?P<minutes>\d+)m)|((?P<seconds>\d+)s)')
```

Comment:
re wasn't imported, so the tests failed, but maybe just `if duration.endswith('d')` something like that?